src/components/RichTextEditor/Plugins/ButtonPlugin/transforms/submitFloatingButton.ts

```diff
@@ -0,0 +1,44 @@
+import { PlateEditor, Value, focusEditor, getPluginOptions } from '@udecode/plate-core';
+import { floatingButtonActions, floatingButtonSelectors } from '../components/FloatingButton/floatingButtonStore';
+import { ButtonPlugin, ELEMENT_BUTTON } from '../createButtonPlugin';
+import { upsertButton } from './index';
+
+/**
+ * Insert button if url is valid.
```

Comment:
could you split it up in functions with the according naming and remove the comments?

Reply:
done. I removed the comment. It's clear what the method does anyway....